src/Plugin/AddPathPlugin.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Common\Plugin;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Add base path to the request URI. Useful for base API URLs like http://domain.com/api.
+ *
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class AddPathPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $host;
+
+    /**
+     * @param UriInterface $host
+     */
+    public function __construct(UriInterface $host)
```

Comment:
no, we do `$host->getPath()` once here to validate, then again where we use it to just get the path part. if we would accept any string, you could create this plugin with "foo bar baz" and we would try to change the request uri to that. or we start to parse urls in here (which might have been validated and parsed before being passed), which is what the Uri model would free us from.